internal/service/withdraw.go

```diff
@@ -47,13 +53,18 @@ func (s WithdrawService) ProcessWithdraw(withdraw model.Withdraw) error {
 	}
 	b.Balance = expectedAfterWithdraw
 	b.SpentAllTime = currentBalance.SpentAllTime + withdraw.Sum
-
-	//TODO надо делать списание и обновление баланса в одной транзакции
-	err := s.repo.Save(&b)
-	if err != nil {
-		return err
-	}
-	err = s.repo.Save(&withdraw)
+	ctx := context.Background()
+	err = s.repo.Atomic(ctx, func(r dao.Repository) error {
+		err = s.repo.SaveBalance(&b)
```

Comment:
ты твой `r` из аргумента не используешь. Тогда должно быть так
```suggestion
		err = r.SaveBalance(&b)
```
ниже аналогично

Reply:
Действительно. Тем самым я не использую транзакцию (
Спасибо, что подсветил